src/components/NoteDetailsPage/NoteDetailsPage.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import styled from 'styled-components';
+import IconButton from '@material-ui/core/IconButton';
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
+import { useDispatch } from 'react-redux';
+import { PAGES_ENUM } from 'components/App/AppConstants';
+import { CHANGE_PAGE_ACTION } from 'components/App/AppActions';
+import { NoteDetailsContainer } from 'components/NoteDetails/NoteDetailsContainer';
+import { Link } from 'react-router-dom';
+
+const Wrapper = styled.div`
+  background-color: #2b2630;
+`;
+
+const BackButtonWrapper = styled.div`
+  color: red !important;
+  padding: 1rem;
+`;
+
+export const NoteDetailsPage = () => {
+  const dispatch = useDispatch();
+
+  const handleClick = () => {
+    dispatch({ type: CHANGE_PAGE_ACTION, data: PAGES_ENUM.NOTES });
```

Comment:
You have deleted code from `AppComponent.js` which used `state.page`, and now `react-router` is used. So why do we still need to dispatch `CHANGE_PAGE_ACTION`, which would update `state.page`, which is not used anywhere?

Reply:
Now we have two actions: `@@router/LOCATION_CHANGE` and `CHANGE_PAGE_ACTION`, so it's basically a logic duplication

![image](https://user-images.githubusercontent.com/6686679/97079742-15b59e80-15ff-11eb-878e-0b08bd9b545b.png)

Moreover, `CHANGE_PAGE_ACTION` is not changing `state` for this case, because the `NoteDetailsButton` never dispatches `CHANGE_PAGE_ACTION`. So when we go from `/notes` to `/note`, `state.page` stays as `/notes`, it never changes to `/note`. But then when we go back from `/note` to `/notes,` we try to update it to `/notes`... That's quite confusing.

If we want to replace our custom page change handling with `react-router`, let's delete everything related to `CHANGE_PAGE_ACTION`? What is the point of keeping both?

Reply:
Please leave that to refactoring branch

Reply:
fixed by https://github.com/DarthVanger/NeuralNotes/pull/167/commits/26d2a17600286be2fa683cb0217232fe490d6fca